lib/gollum/app.rb

```diff
@@ -435,12 +435,12 @@ class App < Sinatra::Base
       end
 
       post '/compare/*' do
-        @file     = encodeURIComponent(params[:splat].first)
```

Comment:
Can you explain why `encodeURIComponent` is no longer necessary? Do we still need to use it elsewhere?

Reply:
I am still struggling to understand this. First, I replaced `redirect_to` with `redirect to` because the latter respects `base_path` as in the other parts in the program. Then I found that `encodeURIComponent` converts `/` to `%2F` in a URL path, which makes the address bar very ugly. But just removing it was a bad idea, sorry, because it is necessary for example when a path contains whitespace and so on. Now I found that ruby's standard `URI.encode()` makes better results in most cases: slashes are left as they are, spaces are converted to `%20`. Why is `encodeURIComponent` implemented in gollum?

Reply:
Now I found that `URI.encode` is obsolete.
https://ruby-doc.org/stdlib-2.7.0/libdoc/uri/rdoc/URI/Escape.html

Reply:
> Now I found that URI.encode is obsolete.

Well, that settles it -- but perhaps the original reason for not using it was that `URI.encode` only allows encoding entire, properly formed URIs? Not sure about that.  But anyway, maybe we can use this: https://apidock.com/ruby/ERB/Util/url_encode ?

I think that may allow us to remove the entire `uri_encode_component.rb` file

Reply:
That sounds really nice. But shall we make that another task? I will revert the things here to some ugly-but-works state to stick to the redirection problem.

Reply:
👍